Formula/synergy-core.rb

```diff
@@ -0,0 +1,376 @@
+           "  It is not adequate to untick the checkbox beside each relevant application. You must\n" \
+           "  remove each application from the list using the '-' button before re-adding them.\n\n" \
+           "  Alternatively, run this command to clear the 'Accessibility' permissions page:\n" \
+           "    sudo tccutil reset Accessibility\n" \
+           "  After running this command, you will need to re-add programs that you want in the list.\n\n" \
+           "  Make sure that you exit Synergy before revoking the 'Accessibility' permission. If\n" \
+           "  Synergy is running when the permission is revoked, you may experience highly anomalous\n" \
+           "  behavior with your mouse and keyboard."
+      if Dir.exist?("/Applications/Synergy.app")
+        s += "\n\n" \
+             "If you previously installed Synergy without using this formula, you may want to\n" \
+             "delete your previous installation to avoid confusion. It appears that you have a\n" \
+             "previous non-brew installation of Synergy at /Applications/Synergy.app. You may\n" \
+             "experience problems granting the 'Accessibility' permission to both copies of\n" \
+             "Synergy at once, even if they aren't running at the same time."
+      end
+      return s
+    end
+  end
+
+  test do
+    port = free_port
+    conf_file = testpath/"synergy.conf"
+    conf_file.write <<~EOS
+      section: screens
+        test-server:
+        test-client:
+      end
+    EOS
+
+    # The Synergy client and server print messages with two formats:
+    #   [2022-04-24T15:18:27] NOTE: started client
+    # and
+    #    2022-04-24 15:18:27.816 synergyc[57673:302843] starting cocoa loop
+    #
+    # In both cases, the message text starts after the first ']' character.
+    log_trim = ->(log_line) { log_line.split("]", 2)[1]&.strip! }
+
+    def kill_and_wait_for(process)
+      Process.kill("KILL", process.pid)
+      Process.wait(process.pid)
+    end
+
+    # In certain cases (discussed below), it is not possible to do any nontrivial
+    # tests. To make sure that Synergy at least compiled successfully, we'll do
+    # some totally trivial checks here. These checks are bad, but they are better
+    # than nothing, which is what would otherwise be possible in certain cases.
+    assert_equal `#{opt_bin}/synergys`, "synergys: no configuration available\n"
+    assert_equal `#{opt_bin}/synergyc`.split("\n")[0], "synergyc: a server address or name is required"
+    synergys_version = `#{opt_bin}/synergys --version`.split("\n")[0]
+    synergyc_version = `#{opt_bin}/synergyc --version`.split("\n")[0]
+    if build.head?
+      version_string = Regexp.quote(version.to_s.split("-")[1])
+      assert_match(/synergys [\-.0-9a-z]*-#{version_string}[a-f0-9]*, protocol version/, synergys_version)
+      assert_match(/synergyc [\-.0-9a-z]*-#{version_string}[a-f0-9]*, protocol version/, synergyc_version)
+    else
```

Comment:
Since head builds are not supported we can just leave this out

Reply:
Done. I removed the special logic for the head build, but I did add a check to avoid testing the version if it's a head build, because even though the head build is not supported, I don't think we need to deliberately make the tests fail.